homeassistant/components/aqara/__init__.py

```diff
@@ -0,0 +1,196 @@
+"""Support for Aqara Smart devices."""
+from __future__ import annotations
+
+import logging
+from typing import NamedTuple
+
+from aqara_iot import (
+    AqaraDeviceListener,
+    AqaraDeviceManager,
+    AqaraHomeManager,
+    AqaraOpenAPI,
+    AqaraOpenMQ,
+    AqaraPoint,
+)
+import requests
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+
+from .const import (
+    AQARA_DISCOVERY_NEW,
+    AQARA_HA_SIGNAL_REGISTER_POINT,
+    AQARA_HA_SIGNAL_UPDATE_ENTITY,
+    AQARA_HA_SIGNAL_UPDATE_POINT_VALUE,
+    CONF_COUNTRY_CODE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    DOMAIN,
+    PLATFORMS,
+)
+from .util import string_dot_to_underline, string_underline_to_dot
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomeAssistantAqaraData(NamedTuple):
+    """Aqara data stored in the Home Assistant data object."""
+
+    device_listener: DeviceListener
+    device_manager: AqaraDeviceManager
+    home_manager: AqaraHomeManager
+    aqara_mqtt_client: AqaraOpenMQ
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Async setup hass config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    api = AqaraOpenAPI(country_code=entry.data[CONF_COUNTRY_CODE])
+
+    try:
+        response = await hass.async_add_executor_job(
+            api.get_auth,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            "",
+        )
+    except requests.exceptions.RequestException as err:
+        raise ConfigEntryNotReady(err) from err
+
+    if response is not True:
+        raise ConfigEntryNotReady("api.get_auth error")
+
+    device_manager = AqaraDeviceManager(api)
+
+    home_manager = AqaraHomeManager(api, device_manager)
+
+    mqtt_client = AqaraOpenMQ()
+    mqtt_client.set_get_config(device_manager.config_mqtt_add)
+    mqtt_client.add_message_listener(device_manager.on_message)
+    mqtt_client.start()
+
+    listener = DeviceListener(hass, device_manager)
+    device_manager.add_device_listener(listener)
```

Comment:
hard to understand what this logic is doing. Does the library have it's own MQTT broker?

Reply:
Hi raman， the library have own MQTTT client.  this logic is setup aqara Integration ,first get all devices from aqara cloud and get the datapoints of these devices, after that will call the Library to subScribe the datapoints use mqtt.